client/src/features/users/Auth/Steps/CreateOrgForm.tsx

```diff
@@ -0,0 +1,284 @@
+];
+
+const SignupSchema = Yup.object().shape({
+  doing_business_as: Yup.string().required('Required'),
+  city: Yup.string().required('Required'),
+  state: Yup.string().required('Required').min(2).max(2),
+  ein: Yup.string()
+    .matches(/^[0-9]\d?-\d{7}$/, 'EIN must match: 99-9999999')
+    .required('Required'),
+  name: Yup.string(),
+  description: Yup.string().required('Required'),
+  website: Yup.string().required('Required'),
+  address: Yup.string().required('Required'),
+  phone: Yup.string()
+    .matches(/^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]\d{3}[\s.-]\d{4}$/, 'EIN must match: 99-9999999')
+    .required('Required'),
+  nonprofit_classification: Yup.string().min(2).required('Required'),
+});
+
+interface Props {
+  triggerNextStep: (step: number) => void;
+  setParentOrg: (createdOrg: Organization) => void;
+  parentOrg: Organization;
+  classes: ClassNameMap<any>;
+}
+const CreateOrgForm = ({
+  setParentOrg = () => {},
+  parentOrg,
+  triggerNextStep = () => {},
+  classes,
+}: Props) => {
+  const [org, setOrg] = React.useState<Organization>(parentOrg);
+  const [triggerEinSearch, setTriggerEinSearch] = React.useState<boolean>(false);
+
+  const orgCreateMutation = useMutation<AxiosResponse<any, any>, Error, Organization, Error>(
+    (newOrg: Organization) => {
+      return axios.post(`http://localhost:3001/api/organizations`, { ...newOrg });
+    },
+    {
+      onSuccess: (data: any) => {
+        triggerNextStep(1);
+        setParentOrg((data as any).data);
+      },
+    },
+  );
+
+  const orgValidateEinQuery = useQuery<
+    AxiosResponse<any, any>,
+    unknown,
+    { ein: string; name: string },
+    string[]
+  >({
+    enabled: triggerEinSearch,
+    queryKey: ['orgValidateEinQuery', org.ein],
+    queryFn: ({ queryKey }) => {
+      const [, ein] = queryKey;
+      return axios.get(`http://localhost:3001/api/organizations/ein/${ein}`);
+    },
+    onSuccess: (res: any) => {
+      setTriggerEinSearch(false);
+      setOrg({ ...org, name: res.data.name });
+    },
+    onError: (res: any) => {
+      setOrg({ ...org, name: '' });
+    },
+    retry: 0,
+  });
+
+  const handleNext = (values: Organization) => {
+    if (org.id === undefined) {
+      setOrg({ ...org });
+      orgCreateMutation.mutate({ ...values, name: org.name });
+    } else {
+      triggerNextStep(1);
+    }
+  };
+
+  return (
+    <React.Fragment>
+      {orgCreateMutation.isLoading ? (
+        <SimpleSnackbar message={`Creating Organization`} />
+      ) : (
+        <>
+          {orgCreateMutation.isError && (
+            <SimpleSnackbar message={`An error occurred ${orgCreateMutation.error}`} />
+          )}
+
+          {orgCreateMutation.isSuccess && <SimpleSnackbar message={`Organization created`} />}
+        </>
+      )}
+      <Formik initialValues={org} validationSchema={SignupSchema} onSubmit={() => {}}>
+        {({ handleChange, values, touched, errors, setFieldTouched, setFieldValue, isValid }) => (
+          <form>
```

Comment:
my thinking is that we shouldn't create an org and then create a user separately
this current pattern allows for just the org (with no user) to be created
so i'd think these 2 pages/steps should be wrapped by 1 `<form>` element that sends 2 simultaneous requests or hits 1 endpoint with all the data and that endpoint handles the 2 db writes
thoughts?